cdp_scrapers/legistar_utils.py

```diff
@@ -109,3 +184,805 @@ def get_legistar_events_for_timespan(
 
     log.debug(f"Collected {len(response)} Legistar events")
     return response
+
+
+def stripped(in_str) -> str:
+    """
+    Return leading and trailing whitespace removed if it is a string
+
+    Parameters
+    ----------
+    in_str: str
+
+    Returns
+    -------
+    str or Any
+        in_str stripped if it is a string
+    """
+    if isinstance(in_str, str):
+        return in_str.strip()
+    return in_str
+
+
+def reduced_list(in_list: List) -> List:
+    """
+    Remove all None items from in_list
+    Return None if in_list is empty
+
+    Parameters
+    ----------
+    in_list : List
+
+    Returns
+    -------
+    List | None
+    """
+    for i in range(len(in_list) - 1, -1, -1):
+        if not in_list[i]:
+            del in_list[i]
+
+    if len(in_list) == 0:
+        in_list = None
+
+    return in_list
+
+
+class LegistarScraper:
+    """
+    Base class for transforming Legistar API data to CDP IngestionModel
+    A given installation must define a derived class and implement get_video_uris()
+
+    Parameters
+    ----------
+    client: str
+        Legistar client name, e.g. "seattle" for Seattle
+
+    Attributes
+    ----------
+    client_name: str
+        Legistar client name
+    MIN_INGESTION_KEYS: Dict[type, List[str]]
+        Keys per IngestionModel used to decide if the given model is empty
+    IGNORED_MINUTE_ITEMS: List[str]
+        Treat EventMinutesItem as None if minutes_item.description
+        containts one of these strings
+    *_pattern: Pattern
+        regex patterns to decide CDP constant from Legistar information
+
+    Methods
+    -------
+    is_legistar_compatible()
+        Check that Legistar API recognizes client name
+    check_for_cdp_min_ingestion(check_days=7)
+        Test if can obtain at least one minimally defined EventIngestionModel
+    get_events(
+        begin=datetime.utcnow() - timedelta(days=2),
+        end=datetime.utcnow()
+    )
+        Main get method that returns Legistar API data as List[EventIngestionModel]
+    get_video_uris(legistar_ev)
+        Must implement in class derived from LegistarScraper
+        if EventVideoPath in Legistar data set is empty
```

Comment:
No need to have these here. These should be documented in each property / method docstring.